### PR TITLE
Refactor Value usage

### PR DESCRIPTION
**Pull request by d-unseductable**
- Rename old `Value` type to `InternalValue`
 - Extract `Value` to a separate struct
 - Implement `From<InternalValue>` for `Value`
 - Move `is_<some_type>` functions to `Value` implementation
   and expose them publicly

@malept, your #13 inspired me to go even further :)

Long story short, I planned to implement `Object.is_nil()` this week and your PR made it much easier. So I decided to refactor the code a bit and move `rb_type()` inside the `Value` struct.

Now `util::rb_type()` function becomes `Value.ty()` and all the `is_fixnum()`, `is_true()` and others become publicly exposed functions of `Value` (`Value.is_fixnum()`). These changes should make it easier to do type checking on higher level inside classes.

What do you think about this?

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -1,15 +1,5 @@
-use ruby_sys::types;
+pub use ruby_sys::types::*;
```
Comment (malept) [inline src/types.rs]: My Python background is making me involuntarily cringe at the `*`.

Comment (d-unseductable) [inline src/types.rs]: 😅 

This file was identical to `ruby-sys`'s `types.rs` + `Callback` type. So I decided just to reexport original `types.rs` here

Comment (malept) [inline src/types.rs]: I don't remember offhand if it's OK (idiomatically speaking) to re-export with a `*` in Rust.

*(Apart from this, I don't have any problems with this PR.)*

Comment (steveklabnik) [inline src/types.rs]: It's not _totally_ frowned upon, but some people feel more strongly than others. Especially with a `pub use`, it can be tricky.

**Comment by d-unseductable** [comment]
Ok guys, I reverted `types` module to previous state. Will spend some time later to remove duplications

**Pull request merged**